apps/api/src/app/admin/admin.service.ts

```diff
@@ -261,6 +261,26 @@ export class AdminService {
         this.prismaService.symbolProfile.count({ where })
       ]);
 
+      const lastMarketPrices = await this.prismaService.marketData.findMany({
+        select: {
+          symbol: true,
+          dataSource: true,
+          marketPrice: true
+        },
+        where: {
+          symbol: { in: assetProfiles.map(({ symbol }) => symbol) },
+          dataSource: { in: assetProfiles.map(({ dataSource }) => dataSource) }
+        },
+        orderBy: { date: 'desc' },
+        distinct: ['symbol', 'dataSource']
+      });
```

Comment:
I opted to fetch all the data before the mapping in one db call to avoid repeated db queries while mapping from `Asset` to `AdminMarketData`. Not 100% sure if this is necessary though, doing it right in the mapping introduces less complexity.